inc/elasticsearch_packages/namespace.php

```diff
@@ -0,0 +1,432 @@
+ * @param string $file The package file path.
+ * @param bool $for_network Whether this is a network level package or not.
+ * @return string|WP_Error
+ */
+function create_package_id( ?string $package_id, string $slug, string $file, bool $for_network = false ) {
+
+	// Check file is an S3 file path.
+	if ( strpos( $file, 's3://' ) !== 0 ) {
+		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		return new WP_Error( 'package_file_missing', sprintf( 'The given package filepath %s is not a valid S3 file path.', $file ) );
+	}
+
+	// Check domain.
+	if ( ! defined( 'ELASTICSEARCH_HOST' ) || ! ELASTICSEARCH_HOST ) {
+		return new WP_Error( 'elasticsearch_host_not_found', 'Could not find an AWS ElasticSearch Service Domain to associate the package with.' );
+	}
+
+	// Get AES client.
+	$client = get_es_client();
+
+	// Get Domain ID from host.
+	$domain = get_elasticsearch_domain();
+
+	// Get a unique package name.
+	// Must be no more than 28 chars.
+	$name = substr( sprintf(
+		'p%s-%s',
+		hash( 'crc32', $domain . $file . time() ),
+		trim( basename( $file, '.txt' ), '-' )
+	), 0, 28 );
+
+	try {
+		// Derive S3 bucket and path.
+		preg_match( '#^s3://([^/]+)/(.*?)$#', $file, $s3_file_parts );
+		$s3_bucket = $s3_file_parts[1];
+		$s3_key = $s3_file_parts[2];
+
+		// Create a new package.
+		$result = $client->createPackage( [
+			'PackageName' => $name, // required.
+			'PackageDescription' => $file,
+			'PackageSource' => [ // required.
+				'S3BucketName' => $s3_bucket,
+				'S3Key' => $s3_key,
+			],
+			'PackageType' => 'TXT-DICTIONARY',
+		] );
+	} catch ( Exception $e ) {
+		return new WP_Error( 'create_package_error', $e->getMessage() );
+	}
+
+	$package = $result['PackageDetails'];
+	$status = $package['PackageStatus'];
+
+	// Set package status.
+	if ( $for_network ) {
+		update_site_option( "altis_search_package_status_{$slug}", $status );
+		update_site_option( "altis_search_package_error_{$slug}", $package['ErrorDetails']['ErrorMessage'] ?? null );
+	} else {
+		update_option( "altis_search_package_status_{$slug}", $status );
+		update_option( "altis_search_package_error_{$slug}", $package['ErrorDetails']['ErrorMessage'] ?? null );
+	}
+
+	$package_id = sprintf( 'analyzers/%s', $package['PackageID'] );
+
+	// Get old package ID to remove if one exists - access options directly to bypass status check.
+	if ( $for_network ) {
+		$existing_package_id = get_site_option( "altis_search_package_{$slug}", null );
+	} else {
+		$existing_package_id = get_option( "altis_search_package_{$slug}", null );
+	}
+
+	// Queue up package status check & association routine.
+	if ( in_array( $status, [ 'AVAILABLE', 'COPYING', 'VALIDATING' ], true ) ) {
+		wp_schedule_event( time(), '1minute', 'altis.search.check_package_status', [
+			$package_id,
+			$slug,
+			$for_network,
+			$existing_package_id,
+		] );
+	} else {
+		return new WP_Error( 'create_package_error', sprintf( 'Package upload error, current status is %s', $status ) );
+	}
```

Comment:
This conditional should be flipped to return early and remove the else branch.
```suggestion
	if ( ! in_array( $status, [ 'AVAILABLE', 'COPYING', 'VALIDATING' ], true ) ) {
		return new WP_Error( 'create_package_error', sprintf( 'Package upload error, current status is %s', $status ) );
	}

	wp_schedule_event( time(), '1minute', 'altis.search.check_package_status', [
		$package_id,
		$slug,
		$for_network,
		$existing_package_id,
	] );
```